content/client-lib-development-guide/features.textile

```diff
@@ -84,6 +84,11 @@ h3(#restclient). RestClient
 * @(RSC8)@ Supports two protocols:
 ** @(RSC8a)@ "MessagePack":http://msgpack.org/ binary protocol (this is the default for environments having a suitable level or support for binary data)
 ** @(RSC8b)@ JSON text protocol (used when @useBinaryProtocol@ option is false)
+** @(RSC8c)@ The library sets the @Accept@ and @Content-Type@ type headers, to reflect whether the client is configured to use a binary or JSON based protocol, to @application/x-msgpack@ or @application/json@ respectively.
+** @(RSC8d)@ Irrespective of the specified @Accept@ header, the library should attempt where possible to process any response body based on the content type indicated in a @Content-Type@ header.
+** @(RSC8e)@ If the library is unable to process the specified content type, then the library processes the response as follows.
+*** @(RSC8e1)@ In the case of a @statusCode@ indicating an error response, the error must be propagated back to the caller with as much information as it is possible to obtain from the response, including the @statusCode@ and, where applicable, an error message containing text from the response body to help to diagnose the cause of the error.
+*** @(RSC8e2)@ In the case of a @statusCode@ indicating a non-error response, an error must be propagated back to the caller with a @statusCode@ of 400, a @code@ of @40013@ and an error message containing the original @statusCode@ and, where applicable, text from the response body, to help to diagnose the cause of the error.
```

Comment:
I don't understand this. Where is "status code indicating a non-error response" defined and under what circumstances would this happen? Is this a 200 OK response and in which case are we just describing an inability at the client-side to parse the response body? I'm really confused by this one.

Reply:
refined https://github.com/ably/docs/pull/888/commits/bfcb292156fecf01102086c64cd951f7b018c8b5

though [my earlier question](https://github.com/ably/docs/pull/888#discussion_r441447075) also applies here.